src/YDialogSpy.cc

```diff
@@ -514,7 +514,12 @@ void YDialogSpy::exec()
 
             continue;
         }
-        else if ( event->widget() == priv->upButton ) priv->moveSelectedUp();
+
+        // check for NULL which is generated for timeout events which can be
+        // triggered by the original dialog
+        if (!event->widget()) continue;
```

Comment:
I don't understand how this helps. If `event->widget()` is NULL then it is not equal to any of the below buttons and nothing happens. What am I missing?